src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -91,8 +91,28 @@ public long voteRecipe(String Id, int voteDiff, Transaction t) {
   }
 
   public List<Tag> getTagsMatchingIds(List<String> Ids) {
+    if (Ids == null || Ids.isEmpty()) {
+      return null;
+    }
     Query tagsQuery = DBUtils.tags().whereIn(DBObject.ID_KEY, Ids);
-    return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
+    QuerySnapshot tags = DBUtils.blockOnFuture(tagsQuery.get());
+    // Must include this if-statement to avoid a null-pointer exception when no tags are followed.
```

Comment:
This comment should avoid mentioning anything about followed tags, as this function's contract is strictly based on the given tag Ids.